web/src/lib/actions/list-navigation.ts

```diff
@@ -1,31 +1,109 @@
 import { shortcuts } from '$lib/actions/shortcut';
+import { tick } from 'svelte';
 import type { Action } from 'svelte/action';
 
-export const listNavigation: Action<HTMLElement, HTMLElement> = (node, container: HTMLElement) => {
-  const moveFocus = (direction: 'up' | 'down') => {
-    const children = Array.from(container?.children);
+interface Options {
+  /**
+   * The container element that with direct children that should be navigated.
+   */
+  container: HTMLElement;
+  /**
+   * The id of the currently selected element.
+   */
+  selectedId: string | undefined;
+  /**
+   * A class name to add to the selected element, if any.
+   */
+  selectedClass?: string;
```

Comment:
I don't like the `selectedClass` modification happening in the action here, it feels hacky. I will move this over to use Svelte contexts for the parent `ButtonContextMenu` to communicate with the `<MenuOption>` in its slot and select the active one.

Reply:
I'm not a huge fan of reusing code/functionality like this. I'd much prefer encapsulating this type of behavior in components themselves.

Reply:
To fix this, I created a new Svelte Action specifically for context menus, so the logic can be shared between the `ButtonContextMenu` and `RightClickContextMenu`. I also changed the mechanism for selecting a menu option to be a Svelte store, because more of a pub/sub model is needed for letting a `MenuOption` know that it has been selected.